### DISCUSSION TITLE
[FR] Model choice not yet documented?

### DISCUSSION THREAD

**Discussion by ozett** [jomjol/AI-on-the-edge-device#1882] [General] [answered]
### The Feature

Updated from v12.0.1 to 13.0.8 
checked the documentation for update steps.
found under configuration a choice for models,
but only 2 of them [in the documentation](https://jomjol.github.io/AI-on-the-edge-device-docs/Choosing-the-Model/).

can i benefit from some models? What are the specs? When to choose which?
( found some hints [on the neural-types information](https://jomjol.github.io/AI-on-the-edge-device-docs/Neural-Network-Types/), it gets confusing..)

| which models|
|:---:|
![image](https://user-images.githubusercontent.com/5937811/213485942-dfe8de78-89d6-4357-8b6d-690926a31ca3.png)

digi-cont-xxx-seems versioning

**Comment by caco3** [comment, marked as answer]
Yes, the documentation is work in progress, there is quite some outdated stuff in it, sadly.

For selecting the model:
The 2 pages you found are what we have.

We usually suggest to just play with the models and see which one works best for you.

**Reply by ozett** [reply]
some minor additions here:
https://github.com/jomjol/AI-on-the-edge-device/discussions/1746

**Comment by ozett** [comment]
ver600 work perfecty, 
compared to ver0570

i log the values of the recognition in grafana, and in the display you have very little points out of the line, 
what means that recognition of the numbers is nearly perfect

|image|
|:---:|
![image](https://user-images.githubusercontent.com/5937811/213863854-d1ffaf93-e3e9-4c51-8208-b50ac6e22cf8.png)

**Comment by forrestfunk** [comment]
If you also use the "previous value" feature in settings, you can get rid of those points out of line too, because they would have produced a "rate too high" or "negative rate" error and would not have been logged to your database.
(Or do you already know this and just tried to test the models?)

**Comment by ozett** [comment]
@forrestfunk 
hey, thanks for commenting.
never looked into this "previous value"-feature, but now i will. 

i eased the curve with "mean" values", but i only wanted to see the "trend",
up to now i could live fine with the seldom points outside the line

**Comment by ozett** [comment]
i looked into the configuration-settings and found that "" is already working. anything else to tune for "pre-value-correction"? 
@forrestfunk : did you mean exactly this?

![image](https://user-images.githubusercontent.com/5937811/213886367-d5483d67-4cfc-4501-8d90-08870bb1dc2a.png)

----------------------------

there seems some automatic funktion additionally in the background, without any manual intervention.
https://jomjol.github.io/AI-on-the-edge-device-docs/Correction%20Algorithm/

**Comment by ozett** [comment]
now i used the "set-prev " from the settings and get a "rate-too-hight" for only 1 counter difference?
think i make an issues with this question...

![image](https://user-images.githubusercontent.com/5937811/213908742-68528ec4-92b0-49d3-9f08-13ecef1116a2.png)

edit: maybe thats related? i will change maxrate from 0.05 to 2 (qm?) and see if it helps

![image](https://user-images.githubusercontent.com/5937811/213908893-0df95963-4c96-49b1-92c0-c306b38d3cb5.png)
https://github.com/jomjol/AI-on-the-edge-device/issues/1670#issuecomment-1363213908

**Reply by caco3** [reply]
is it right that the `pre` value was on `24370` and the `raw` value on `24371`?
If those numbers a re right you had an increase of `1.0` since the last round!
Then obviously the `maxRateValue` mit be `>1`!

**Reply by ozett** [reply]
yes. the counter counts quibic meter of gas, only full qm.
so 1 step is the possible minimal change of the counter.

would be nice to have there an automatic adjustment of the `maxRateValue` to change the default into something reasonable,
based on the calculated `rate` . or maybe directly adjust `maxRateValue`  to a suitable value within the first setup on defining the `meter`-type (gas/electric/water, full, float)?

**Reply by caco3** [reply]
automatics are always dangerous because they are hard to test and might do things which are unwanted/unexpected.
 I think it is better when a user learns to understand those parameters and sets them as needed.
There are so many different use cases that it is impossible to have a setup that fits all.
